### ISSUE TITLE
Some Switches missing in the project

### ISSUE THREAD

**Issue by StevieWonder99a** [maciej-or/hikvision_next#260]
Hi, at first - thank you for your work and the integration in Home Assistant.

I miss a few features.
I use Alamo to control some things with one click.

But - it is not possible to do something like this:

if iam goint out, absend - the cameras should enable the alarm-siren/flashlight.
When i come back - disable these features.
Motion detection runs all the day, no problem. But the alarming part should only be aktivated when i use a switch / automation / Scene. But all switches in the project can only make the basics on / off. Motion , line crossing detection, area going in and out detection, but not the features i need. Start and stop the alarm siren or something like this.

HikVision DS-2CD2386G2-ISU/SL
![cam](https://github.com/user-attachments/assets/cdb4882f-f95c-4772-abae-b36edfd40136)
![HA](https://github.com/user-attachments/assets/75133c0b-bb68-4004-865c-d11905c82525)

**Comment by maciej-or** [comment]
Yea, many people have asked about the siren switches. Unfortunately I do not have access to this camera and I'm not able to add this basing on documentation only.

**Comment by StevieWonder99a** [comment]
What do you need? A shared connect with hik-connect ? I get a new cam next week and i could leave it out of service for a few days..Am 27.11.2024 um 19:29 schrieb Maciej ***@***.***>:﻿
Yea, many people have asked about the siren switches. Unfortunately I do not have access to this camera and I'm not able to add this basing on documentation only.

**Comment by maciej-or** [comment]
Hik-connect uses different protocol, wont be useful.
I would need diagnostic data from your camera, then follow this thread 
https://community.home-assistant.io/t/hikvision-cameras-trigger-audio-alarm-output-possible/545886/8
and check if mentioned requests work for you.
You can use `ISAPI request action` instead of curl

![image](https://github.com/user-attachments/assets/001a98d9-32d6-47c8-93dc-76c0997bc2da)


Basically if you are familiar with automations you can create a switch yourself that toggles siren even with current version of the integration.

**Comment by maciej-or** [comment]
#114

**Comment by StevieWonder99a** [comment]
i will see if it works. But - i like functions in HA out of the box. Here are more then 20 different manufacturer for different hardware. if i would use the api and everything by myself - i need 2 people who control the code and api changes. So i am happy with a lot of integrations in home assistant und updates if something changed. Home assistant is for my a big project at home, everything with lan/wifi is under control with lovely dashboards. It is to much for individual programming.

**Comment by maciej-or** [comment]
I just need a confirmation if those requests work, then I could add them to the integration. I'm able to simulate any Hikvision device in test environment but it requires specific XML responses.

**Comment by StevieWonder99a** [comment]
> Hik-connect uses different protocol, wont be useful. I would need diagnostic data from your camera, then follow this thread https://community.home-assistant.io/t/hikvision-cameras-trigger-audio-alarm-output-possible/545886/8 and check if mentioned requests work for you. You can use `ISAPI request action` instead of curl
> 
> ![image](https://private-user-images.githubusercontent.com/38075949/390547816-001a98d9-32d6-47c8-93dc-76c0997bc2da.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzMxMzI1NzksIm5iZiI6MTczMzEzMjI3OSwicGF0aCI6Ii8zODA3NTk0OS8zOTA1NDc4MTYtMDAxYTk4ZDktMzJkNi00N2M4LTkzZGMtNzZjMDk5N2JjMmRhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMDIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjAyVDA5Mzc1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTM1MzMwMGY1MDNhMjA1ZjM4MGM2NmFkZDI3ZjNhZDViZTljZmRlMDQ2ZjVmNzhhZjk0ZDY0NzVjMTI3ZjBmZGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.a0sp9TrmSl5Ozg2wNhFK-_lkj60rYhtfYUHpLRQYb8w)
> 
> Basically if you are familiar with automations you can create a switch yourself that toggles siren even with current version of the integration.

Where to send the diagnostic Data ?

**Comment by maciej-or** [comment]
see https://github.com/maciej-or/hikvision_next/wiki/How-to-capture-ISAPI-response
just paste responses here

**Comment by StevieWonder99a** [comment]
> see https://github.com/maciej-or/hikvision_next/wiki/How-to-capture-ISAPI-response just paste responses here

[DS-2CD2386G2-ISU_SL.txt](https://github.com/user-attachments/files/17975922/DS-2CD2386G2-ISU_SL.txt)

**Comment by StevieWonder99a** [comment]
![image](https://github.com/user-attachments/assets/8fbe1d77-e01f-4f6f-b578-685caecc74fe)

i got a EA-1-Alarm on Hik-Connect when i send the code. So the system is alive ...

**Comment by StevieWonder99a** [comment]
> see https://github.com/maciej-or/hikvision_next/wiki/How-to-capture-ISAPI-response just paste responses here

[DS-2CD2386G2-ISU_SL (1).txt](https://github.com/user-attachments/files/17976140/DS-2CD2386G2-ISU_SL.1.txt)

**Comment by StevieWonder99a** [comment]
Hey, i got another idea. I set the camera switch for E-A-Alarm to N.O in the cameras and activated the flashlight.

Now, i got an automation. One of the cameras gets a line crossing alert - HA starts to send a E-A Alarm Input to all the cameras, and 10 minutes later, it stops the E-A Alarm and the flashlights are switching off. For me, this is what i want.

**Comment by zx8** [comment]
@maciej-or Sorry to bug you, but is there any more information needed that would help you to implement this feature? I am happy to provide logs etc
